GameOfLife/GameOfLife/Services/FileIO.cs

```diff
@@ -0,0 +1,228 @@
+        private IEngine _engine;
+        private int _numberOfFiles;
+        private string _filePath = @"C:\GameOfLife_SavedGames\";
+        private string[] _stringField;
+        private List<string> _stringList = new List<string>();
+        private bool _fileReadingError = false;
+        private bool _fileLoaded = false;
+        private bool _noSavedGames = false;
+        public bool FileReadingError
+        {
+            get => _fileReadingError;
+            set => _fileReadingError = value;
+        }
+        public string FilePath
+        {
+            get => _filePath;
+        }
+        public bool FileLoaded
+        {
+            get => _fileLoaded;
+            set => _fileLoaded = value;
+        }
+        public bool NoSavedGames
+        {
+            get => _noSavedGames;
+            set => _noSavedGames = value;
+        }
+        public int NumberOfFiles
+        {
+            get => _numberOfFiles;
+        }
+
+        public void Injection(IRender render, IInputController inputController, IEngine engine)
+        {
+            _render = render;
+            _inputController = inputController;
+            _engine = engine;
+        }
+
+        /// <summary>
+        /// Method to check if the directory exists. If it doesn't, the method creates one.
+        /// </summary>
+        /// <param name="filePath">The location of the folder.</param>
+        private void EnsureDirectoryExists(string filePath)
+        {
+            FileInfo fileInfo = new(filePath);
+            if (!fileInfo.Directory.Exists)
+            {
+                Directory.CreateDirectory(fileInfo.DirectoryName);
+            }
+        }
+
+        /// <summary>
+        /// Method which converts 2-dimensional array to a 1-dimensional in order to minimize the number of 'write' operations to a file.
+        /// </summary>
+        /// <param name="gameField">An instance of the GameFieldModel class that stores the game field and its properties.</param>
+        /// <returns>Returns new 1-dimensional array of game field rows.</returns>
+        private string[] ConvertGameFieldToArrayOfRows(GameFieldModel gameField)
+        {
+            _stringField = new string[gameField.Width];
+
+            for (int x = 0; x < gameField.Length; x++)
+            {
+                for (int y = 0; y < gameField.Width; y++)
+                {
+                    _stringField[y] = _stringField[y] + gameField.GameField[x, y] + " ";
+                }
+            }
+            return _stringField;
+        }
+
+        /// <summary>
+        /// Method that takes the list of the game field cells and converts it to an array of the game field.
+        /// </summary>
+        /// <param name="inputList">List of the game field cells.</param>
+        /// <returns>Returns an instance of the GameFieldModel class.</returns>
+        private GameFieldModel ConvertListOfRowsToGameField(List<string> inputList)
+        {
+            int x = 0;
```

Comment:
More descriptive names needed for variables x and y.

Reply:
Changed the names of variables x and y.

Reply:
It is better but awkward.
May be
int horizontalX = 0;    =>   column or horizontalOffset or xCoordinate
and 
int verticalY = 0;    =>   row or verticalOffset or yCoordinate

Reply:
Yes, I agree. At first I changed x and y to horizontalCoordinate and verticalCoordinate, but then I thought that it was a bit too long and changed. 

xCoordinate and yCoordinate looks good, changed to such names.